firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -395,32 +421,42 @@ AppDistributionReleaseInternal getCachedNewRelease() {
     if (newRelease.getReleaseNotes() != null && !newRelease.getReleaseNotes().isEmpty()) {
       message.append(String.format("\n\nRelease notes: %s", newRelease.getReleaseNotes()));
     }
-    updateDialog.setMessage(message);
+    updateConfirmationDialog.setMessage(message);
 
-    updateDialog.setButton(
+    updateConfirmationDialog.setButton(
         AlertDialog.BUTTON_POSITIVE,
         context.getString(R.string.update_yes_button),
         (dialogInterface, i) -> showUpdateDialogTask.setResult(null));
 
-    updateDialog.setButton(
+    updateConfirmationDialog.setButton(
         AlertDialog.BUTTON_NEGATIVE,
         context.getString(R.string.update_no_button),
-        (dialogInterface, i) ->
-            showUpdateDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.UPDATE_CANCELED, Status.INSTALLATION_CANCELED)));
+        (dialogInterface, i) -> updateConfirmationDialogClosedCallback(showUpdateDialogTask));
 
-    updateDialog.setOnCancelListener(
-        dialogInterface ->
-            showUpdateDialogTask.setException(
-                new FirebaseAppDistributionException(
-                    ErrorMessages.UPDATE_CANCELED, Status.INSTALLATION_CANCELED)));
+    updateConfirmationDialog.setOnCancelListener(
+        dialogInterface -> updateConfirmationDialogClosedCallback(showUpdateDialogTask));
 
-    updateDialog.show();
+    updateConfirmationDialog.show();
 
     return showUpdateDialogTask.getTask();
   }
 
+  private void setRemakeDialogsToFalse() {
```

Comment:
Since we're just using this in one place, and it's already being called from a synchronized block, we might as well just inline it.